### PR TITLE
Console game

### PR DESCRIPTION
**Pull request by YuriAICruz**

### CODE REVIEW

File: GameConsole/wInclude.cpp [hunk 1]
```diff
@@ -0,0 +1 @@
+﻿#include "wInclude.h"
```
Comment (psydack) [inline GameConsole/wInclude.cpp:1]: Are you going to use it later?

File: GameConsole/screens/settingsMenu.h [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+            for (std::pair<const int, eventBus::delegate<transitionData>> transitionMap : transitionsMap)
+            {
+                size_t id = events->subscribe<transitionData>(transitionMap.first, transitionMap.second.action);
+                transitionsMap[transitionMap.first].uid = id;
+            }
+
+            players.resize(2);
+            for (int i = 0, n = players.size(); i < n; ++i)
+            {
+                std::stringstream ss;
```
Comment (psydack) [inline GameConsole/screens/settingsMenu.h:58]: Creating one each iteration is better than reusing?

File: GameConsole/screens/settingsMenu.h [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+        settingsMenu(std::shared_ptr<renderer::renderer> rdr, std::shared_ptr<eventBus::eventBus> events)
+            : IScreen(rdr, events)
+        {
+            for (std::pair<const int, eventBus::delegate<transitionData>> transitionMap : transitionsMap)
+            {
+                size_t id = events->subscribe<transitionData>(transitionMap.first, transitionMap.second.action);
+                transitionsMap[transitionMap.first].uid = id;
+            }
+
+            players.resize(2);
```
Comment (psydack) [inline GameConsole/screens/settingsMenu.h:55]: Why 2?

File: GameConsole/screens/settingsMenu.h [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+            for (int i = 0, n = players.size(); i < n; ++i)
+            {
+                std::stringstream ss;
+                ss << "Player 0" << (i + 1);
+                players[i] = ss.str();
+            }
+            playersButtons.resize(12);
```
Comment (psydack) [inline GameConsole/screens/settingsMenu.h:62]: Why 12?

File: GameConsole/wInclude.cpp (hunk 1 above)
Comment (YuriAICruz) [reply GameConsole/wInclude.cpp:1]: no idea yet, left just in case

File: GameConsole/screens/settingsMenu.h (hunk 2 above)
Comment (YuriAICruz) [reply GameConsole/screens/settingsMenu.h:58]: no, this one needs a refactoring for sure

File: GameConsole/screens/settingsMenu.h (hunk 3 above)
Comment (YuriAICruz) [reply GameConsole/screens/settingsMenu.h:55]: magic numbers gotta love them :), I'll create a const for it

**Pull request merged**